test/savefiles_tests.jl

```diff
@@ -307,6 +307,18 @@ end
     rm(path; recursive = true, force = true)
 end
 
+# Testing proper filenames when default_prefix was modified. See https://github.com/JuliaDynamics/DrWatson.jl/issues/392
+@testset "@produce_or_load with default_prefix modified" begin
+    path = mktempdir()
+
+    DrWatson.default_prefix(ntuple::NamedTuple) = "Prefix_"
+
+    sim, path = produce_or_load(f, simulation, "")
+    @test path == savename(simulation)
```

Comment:
Please don't infect the `NamedTuple` here, because this could mess up tests after this one (if some are added in the future). Instead make a dummy type like
```julia
struct Dummy
x
y
end

simulation = Dummy(1,2)
DrWatson.default_prefix(...)
```